web/src/pill_typeahead.ts

```diff
@@ -129,6 +129,45 @@ export function set_up_stream(
     });
 }
 
+export function set_up_user_group(
+    $input: JQuery,
+    pills: user_group_pill.UserGroupPillWidget,
+    opts: {
+        user_group_source: () => UserGroup[];
+    },
+): void {
+    const bootstrap_typeahead_input: TypeaheadInputElement = {
+        $element: $input,
+        type: "contenteditable",
+    };
+    new Typeahead(bootstrap_typeahead_input, {
+        dropup: true,
+        source(_query: string): UserGroupPillData[] {
+            return opts
+                .user_group_source()
```

Comment:
Since we call this function only from the invite UI, `user_group_source` is always expected. If we use this at other places, we can make this optional and instead call the default function which includes all groups.